scripts/Muon/GUI/Common/contexts/muon_gui_context.py

```diff
@@ -53,3 +61,11 @@ def period_string(self, run=None):
             return '+'.join([str(period) for period in summed_periods]) + '-' + '-'.join([str(period) for period in subtracted_periods])
         else:
             return '+'.join([str(period) for period in summed_periods])
+
+    def remove_workspace_by_name(self, workspace_name):
+        try:
+            if self['DeadTimeTable'].name() == workspace_name:
+                self.pop('DeadTimeTable')
+                self['DeadTimeSource'] = 'FromFile'
```

Comment:
when would we get to this line? It might play a part in the deadtime table bug.

Reply:
This line is for the case where the user manually deletes a deadtime table from the ADS. In this case we need to remove it here as it is no longer available to use in calculations.